src/RA_Defs.h

```diff
@@ -151,49 +185,124 @@ typedef DWORD			ARGB;
 
 //namespace RA
 //{
-template<typename T>
-static inline const T& RAClamp(const T& val, const T& lower, const T& upper)
+
+_NODISCARD inline constexpr bool operator==(_In_ const RECT& a, _In_ const RECT& b) noexcept
 {
-    return(val < lower) ? lower : ((val > upper) ? upper : val);
+    return{ a.left   == b.left   and
+            a.top    == b.top    and
+            a.right  == b.right  and
+            a.bottom == b.bottom };
 }
 
-class RARect : public RECT
+// What should we do for less-than? Compare the area, quadrants?
+
+// We can make this into a literal type by removing inheritance
+class RARect
 {
 public:
-    RARect() {}
-    RARect(LONG nX, LONG nY, LONG nW, LONG nH)
+    // Constructors are similar to SetRect
+    inline constexpr RARect(_In_ const RECT& rect) noexcept : m_Rect{ rect } { }
+    inline constexpr RARect(_In_ const POINT& XY, _In_ const POINT& WH) noexcept :
+        m_Rect{ XY.x,         // left
+                XY.y,         // top
+                XY.x + WH.x,  // right
+                XY.y + WH.y } // bottom
+    {
+    }
+
+    inline constexpr RARect(_In_ LONG nX, _In_ LONG nY, _In_ LONG nW, _In_ LONG nH) noexcept :
+        m_Rect{ nX,       // left
+                nY,       // top
+                nX + nW,  // right
+                nY + nH } // bottom
     {
-        left = nX;
-        right = nX + nW;
-        top = nY;
-        bottom = nY + nH;
     }
 
+    RARect(const RARect&) = delete;
+    RARect& operator=(const RARect&) = delete;
+    inline constexpr RARect(RARect&&) noexcept = default;
+    _NODISCARD inline constexpr RARect& operator=(RARect&&) noexcept = default;
+
+    inline constexpr RARect() noexcept = default; 
+
+    _NODISCARD explicit constexpr operator RECT() const noexcept { return m_Rect; }
+    _NODISCARD explicit constexpr operator bool() const noexcept { return std::empty(*this); }
+    // This one can't be const
+    _NODISCARD explicit constexpr operator LPRECT() noexcept { return &m_Rect; }
+
 public:
-    inline int Width() const { return(right - left); }
-    inline int Height() const { return(bottom - top); }
+    _NODISCARD inline constexpr LONG Width() const noexcept { return{ m_Rect.right - m_Rect.left }; }
+    _NODISCARD inline constexpr LONG Height() const noexcept { return{ m_Rect.bottom - m_Rect.top }; }
+
+    _NODISCARD inline constexpr LONG Area() const noexcept { return{ Width()*Height() }; }
+    _NODISCARD inline constexpr bool empty() const noexcept { return{ Area() == 0L }; }
```

Comment:
It's lower case to be compatible with `std::empty` (which is a constant expression, most container's empty method is not a constant expression). Also for a lot of other classes we can make them compatible with `ranged for` be supplying "`begin`" and "`end`" functions like `CodeNotes`, `ConditionSet` or `MemBookmarks`, or any Container based classed.